### PR TITLE
Use cloud_reg_addrs option and do not manually reset nodeaddr on power_down

### PR DESCRIPTION
**Pull request by rexcsn**
* Using cloud_reg_addrs and let slurm reset nodeaddr automatically. This option is available in Slurm 20.11.
* Keep reset nodes for inactivate partitions because nodes are not power_down in this case

Signed-off-by: Rex <shuningc@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/slurm_plugin/clustermgtd.py [hunk 1]
```diff
@@ -798,8 +798,6 @@ def _handle_powering_down_nodes(self, slurm_nodes, private_ip_to_instance_map):
                 powering_down_nodes.append(node)
 
         if powering_down_nodes:
-            log.info("Resetting powering down nodes: %s", print_with_count(powering_down_nodes))
-            reset_nodes(nodes=[node.name for node in powering_down_nodes])
```
Comment (demartinofra) [inline src/slurm_plugin/clustermgtd.py:802]: when is this resetting happening? as soon as the powering down is initiated? If not are there some downstream consequences if we don't reset it here and Slurm hasn't reset this value yet?

Comment (demartinofra) [reply src/slurm_plugin/clustermgtd.py:802]: let's also update the docstring by mentioning that we don't reset the nodeaddr since Slurm takes care of it

Comment (rexcsn) [reply src/slurm_plugin/clustermgtd.py:802]: Docstring updated. Made modification to code to account for the fact that nodeaddr are not reset for powering_down nodes.

**Review by demartinofra — COMMENTED** [review]
how are now powering down nodes handled?do they result as orphaned instances? If so we might want to adjust the logging message making it clear that we are terminating not only unexpected orphaned instances but also powering down nodes.

File: src/slurm_plugin/clustermgtd.py [hunk 2]
```diff
@@ -790,16 +795,16 @@ def _handle_powering_down_nodes(self, slurm_nodes, private_ip_to_instance_map):
...
         """
         powering_down_nodes = []
         for node in slurm_nodes:
             if not node.is_static and node.is_nodeaddr_set() and (node.is_power() or node.is_powering_down()):
                 powering_down_nodes.append(node)
 
         if powering_down_nodes:
-            log.info("Resetting powering down nodes: %s", print_with_count(powering_down_nodes))
-            reset_nodes(nodes=[node.name for node in powering_down_nodes])
             instances_to_terminate = ClusterManager._get_backing_instance_ids(
```
Comment (demartinofra) [inline src/slurm_plugin/clustermgtd.py:808]: can we still retrieve the mapping here?when is the nodeaddr unset by Slurm?

**Review by demartinofra — COMMENTED** [review]
how are now powering down nodes handled?do they result as orphaned instances? If so we might want to adjust the logging message making it clear that we are terminating not only unexpected orphaned instances but also powering down nodes.

**Review by demartinofra — DISMISSED** [review]

File: src/slurm_plugin/clustermgtd.py (hunk 2 above)
Comment (rexcsn) [reply src/slurm_plugin/clustermgtd.py:813]: Added some comments to clarify this

**Review by demartinofra — APPROVED** [review]

**Pull request merged**